fe/src/main/java/org/apache/doris/load/loadv2/SparkEtlJobHandler.java

```diff
@@ -0,0 +1,259 @@
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.apache.spark.launcher.SparkAppHandle;
+import org.apache.spark.launcher.SparkAppHandle.Listener;
+import org.apache.spark.launcher.SparkAppHandle.State;
+import org.apache.spark.launcher.SparkLauncher;
+
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.List;
+import java.util.Map;
+
+public class SparkEtlJobHandler {
+    private static final Logger LOG = LogManager.getLogger(SparkEtlJobHandler.class);
+
+    private static final String JOB_CONFIG_DIR = PaloFe.DORIS_HOME_DIR + "/temp/job_conf";
+    private static final String JOB_CONFIG_FILE = "jobconfig.json";
+    private static final String APP_RESOURCE = PaloFe.DORIS_HOME_DIR + "/lib/palo-fe.jar";
+    private static final String MAIN_CLASS = "org.apache.doris.load.loadv2.etl.SparkEtl";
+    private static final String ETL_JOB_NAME = "doris__%s";
+    // hdfs://host:port/outputPath/dbId/loadLabel/PendingTaskSignature
+    private static final String ETL_OUTPUT_PATH = "%s%s/%d/%s/%d";
+    // http://host:port/api/v1/applications/appid/jobs
+    private static final String STATUS_URL = "%s/api/v1/applications/%s/jobs";
+
+    public static final String NUM_TASKS = "numTasks";
+    public static final String NUM_COMPLETED_TASKS = "numCompletedTasks";
+
+    class SparkAppListener implements Listener {
+        @Override
+        public void stateChanged(SparkAppHandle sparkAppHandle) {}
+
+        @Override
+        public void infoChanged(SparkAppHandle sparkAppHandle) {}
+    }
+
+    public SparkAppHandle submitEtlJob(long loadJobId, String loadLabel, String sparkMaster,
+                                       Map<String, String> sparkConfigs, String jobJsonConfig) throws LoadException {
+        // check outputPath exist
+
+        // create job config file
+        String configDirPath = JOB_CONFIG_DIR + "/" + loadJobId;
+        String configFilePath = configDirPath + "/" + JOB_CONFIG_FILE;
+        try {
+            createJobConfigFile(configDirPath, configFilePath, loadJobId, jobJsonConfig);
+        } catch (LoadException e) {
+            return null;
+        }
+
+        // spark cluster config
+        SparkLauncher launcher = new SparkLauncher();
+        launcher = launcher.setMaster(sparkMaster)
+                .setAppResource(APP_RESOURCE)
+                .setMainClass(MAIN_CLASS)
+                .setAppName(String.format(ETL_JOB_NAME, loadLabel))
+                .addAppArgs(configFilePath);
+        for (Map.Entry<String, String> entry : sparkConfigs.entrySet()) {
+            launcher = launcher.setConf(entry.getKey(), entry.getValue());
+        }
+
```

Comment:
1 support add hadoop queue
2 support user specify jars
3 support specify hive-site.xml
```suggestion
launcher.addSparkArg("--queue", "hadoop-queue");
launcher.addSparkArg("--jars", "a.jar,b.jar,c.jar");
launcher.addSparkArg("--files", "/spark-2.2/conf/hive-site.xml");
```